word_nerd/users/serializers.py

```
@@ -0,0 +1,37 @@
+from django.contrib.auth.models import User
+import django.contrib.auth.password_validation as django_validators
+from django.core import exceptions
+from rest_framework import serializers
+
+
+class RegisterSerializer(serializers.ModelSerializer):
+    password2 = serializers.CharField(write_only=True)
+
+    class Meta:
+        model = User
+        fields = ['username', 'password', 'password2']
+
+    def validate(self, data):
+        password = data['password']
+        confirmed_password = data.pop('password2')
+
+        errors = {}
+        if password != confirmed_password:
+            raise serializers.ValidationError(f"Passwords don't match")
+        try:
+            # validate the password and catch errors
+            django_validators.validate_password(
+                password=password, user=User(data['username'], data['password']))
+        except exceptions.ValidationError as e:
+            errors['password'] = list(e.messages)
+        if errors:
+            raise serializers.ValidationError(errors)
+        return data
+
+    def create(self, validated_data):
+        try:
+            user = User.objects.create_user(username=validated_data['username'],
+                                            password=validated_data['password'])
+            return user
+        except Exception as e:
+            return e
```

It's never good to catch all exceptions as `except Exception as e:`. You need to figure out which specific exception this function `create_user` raises and catch that. Also, You usually never `return` exceptions but rather `raise` them. In this case, if you simply write
```python
except Exception as e:
    raise e
```
this is the same as not using a `try except` statement so you can simply delete it